src/server/pfs/server/driver.go

```diff
@@ -3401,7 +3399,8 @@ func getTreeRange(ctx context.Context, objClient obj.Client, path string, prefix
 
 // getTreeForFile is like getTreeForCommit except that it can handle open commits.
 // It takes a file instead of a commit so that it can apply the changes for
-// that path to the tree before it returns it.
+// that path to the tree before it returns it. The returned hash tree is not in
+// the treeCache and must be cleaned up by the caller.
 func (d *driver) getTreeForFile(pachClient *client.APIClient, file *pfs.File) (hashtree.HashTree, error) {
```

Comment:
`getTreeForFile` and `getTreeForOpenCommit` now mention that the tree they return must be destroyed.  I attempted to change these to a callback-based (e.g.) `withTreeForFile` where the cleanup would be builtin and the hashtree scoped to a callback, but the callsites didn't lend themselves to it and it would have been a dangerous, sweeping change.  Since this code isn't expected to be much longer-lived, I think this will do for now.